library/ceph_volume.py

```diff
@@ -603,6 +608,8 @@ def run_module():
             module, activate_osd())
 
     elif action == 'zap':
+        if module.params.get('osd_fsid', None) and module.params.get('osd_id', None):
+            fatal("Choose either osd_fsid or osd_id parameter.", module)
```

Comment:
This could be replaced with a combinaison of `mutually_exclusive` and `required_if` during the `AnsibleModule` call.

Something like (but untested):

```python
        mutually_exclusive=[
            ('data', 'osd_id'),
            ('data', 'osd_fsid'),
            ('osd_fsid', 'osd_id'),
        ],
        required_if=[
            ('action', 'zap', ('data', 'osd_fsid', 'osd_id'))
        ],
```

Reply:
yes maggle